proto/reaction.proto

```diff
@@ -318,11 +349,40 @@ message Vessel {
       CUSTOM = 1;
       NONE = 2;
       OVEN_DRIED = 3;
+      FLAME_DRIED = 4;
+      EVACUATED_BACKFILLED = 5;
+      PURGED = 6;
     }
   }
-  VesselPreparation.VesselPreparationType preparation = 5;
+  repeated VesselPreparation.VesselPreparationType preparations = 5;
   string preparation_details = 6;
-  Volume volume = 7;  // Size (volume) of the vessel.
+  message VesselAttachment {
+    enum VesselAttachmentType {
+      UNSPECIFIED = 0;
+      NONE = 1;
+      CUSTOM = 2;
+      SEPTUM = 3;
+      CAP = 4;
+      REFLUX_CONDENSER = 6;
+      VENT_NEEDLE = 7;
+      DEAN_STARK = 8;
+      VACUUM_TUBE = 9;
+      ADDITION_FUNNEL = 10;
+      DRYING_TUBE = 11;
+      ALUMINUM_FOIL = 12;
+      THERMOCOUPLE = 13;
+      BALLOON = 14;
+      GAS_ADAPTER = 15;  // E.g. a ground-glass adapter with a gas line.
+      PRESSURE_REGULATOR = 16;
+      RELEASE_VALVE = 17;
+    }   
+  }
+  repeated VesselAttachment.VesselAttachmentType attachments = 7;
+  // Include attachment specifications, e.g., "rubber" for SEPTUM,
+  // "Teflon" for CAP, "water jacket" for REFLUX_CONDENSER, etc.
+  // Also include sealing details, e.g., "electrical tape", "parafilm", etc.
+  string attachments_details = 8;
```

Comment:
I'd prefer:

```proto
message VesselAttachment {
  enum VesselAttachmentType {
    ...
  }
  VesselAttachmentType type = 1;
  string details = 2;
}
repeated VesselAttachment attachments = 7;
```

Reply:
I agree, I like this logic. This is actually how I have it in my fork...I may have missed a change? So based on the conversation, should _all_ sub-messages be changed to this structure?

Reply:
That's on me, Mike, as I had changed it to be consistent with our non-repeated fields. In light of our discussion, all _repeated_ fields should use the structure that Steve commented here.

Reply:
No worries, thanks. Updating some other ones in the specific conditions messages now.